### ISSUE TITLE
RCS1109 for implicit operator

### ISSUE THREAD

**Issue by jnyrup** [JosefPihrt/Roslynator#199]
```c#
public class DTO
{
    public static implicit operator DTO(Myclass e) => new DTO();
}

public class Myclass { }
```

```c#
var list = new List<Myclass>() { new Myclass() };
var list1 = list.Select(e => (DTO)e).ToList(); // triggers RCS1109
var list2 = list.Cast<DTO>().ToList(); // <-- throws invalid cast exceptions
```

This seems related to #186

**Comment by JosefPihrt** [comment]
Fixed with #186

**Comment by jnyrup** [comment]
I'm sorry. I forgot to pay to attention to the timestamp of the other issue.
